src/trel_dnssd/trel_dnssd.cpp

```diff
@@ -91,6 +91,9 @@ TrelDnssd::TrelDnssd(Ncp::RcpHost &aHost, Mdns::Publisher &aPublisher)
 void TrelDnssd::Initialize(std::string aTrelNetif)
 {
     mTrelNetif = std::move(aTrelNetif);
+    // Reset mTrelNetifIndex to 0 so that When this function is called with a different aTrelNetif
+    // than the current mTrelNetif, CheckTrelNetifReady() will update mTrelNetifIndex accordingly.
```

Comment:
```suggestion
    // Reset mTrelNetifIndex to 0 so that when this function is called with a different aTrelNetif
    // than the current mTrelNetif, CheckTrelNetifReady() will update mTrelNetifIndex accordingly.
```